### DISCUSSION TITLE
Inertia - Status code on exception

### DISCUSSION THREAD

**Discussion by Xstoudi** [adonisjs/core#4563] [Help] [answered]
Hey, the [inertia doc](https://inertiajs.com/error-handling) mention this:

```php
use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;
use Inertia\Inertia;

->withExceptions(function (Exceptions $exceptions) {
    $exceptions->respond(function (Response $response, Throwable $exception, Request $request) {
        if (! app()->environment(['local', 'testing']) && in_array($response->getStatusCode(), [500, 503, 404, 403])) {
            return Inertia::render('Error', ['status' => $response->getStatusCode()])
                ->toResponse($request)
                ->setStatusCode($response->getStatusCode());
        } elseif ($response->getStatusCode() === 419) {
            return back()->with([
                'message' => 'The page expired, please try again.',
            ]);
        }

        return $response;
    });
})
```

What would be the recommended way to return, for example, a 404 status code?

**Comment by Julien-R44** [comment, marked as answer]
If you're using the inertia starter-kit then it's handled automatically for you in the Exception Handler. See there : https://github.com/adonisjs/inertia-starter-kit/blob/main/app/exceptions/handler.ts#L24

If you throw an exception with a 404 status code, your error page will be rendered with Inertia. For example, this is the case with Lucid's `findxxOrFail` methods: 

```ts
export default class UsersController {
  async index({ inertia, response }: HttpContext) {
    const user = await User.findOrFail(1212121);

    // ...
  }
}
```

If the user is not found, then : 
- The exception [E_ROW_NOT_FOUND](https://github.com/adonisjs/lucid/blob/2a2bd22c2261f2548f4d794131b4b86fcdaea829/src/errors.ts#L48) will be thrown. This same E_ROW_NOT_FOUND exception uses a status code `404` as seen in the linked code.
- We enter the exception handler with this exception that has just been thrown
- The exception handler will apply the [status page rendering logic](https://docs.adonisjs.com/guides/basics/exception-handling#status-pages) according to the status code
- Then [we finally call the `inertia.render`](https://github.com/adonisjs/inertia-starter-kit/blob/main/app/exceptions/handler.ts#L24) which will render your error page via inertia
